cylc/flow/network/resolvers.py

```diff
@@ -338,42 +353,47 @@ def __init__(self, data, **kwargs):
-                args.get('event_message'),
-                args.get('event_id')))
-        if command == 'put_messages':
-            messages = args.get('messages', [])
-            for severity, message in messages:
-                self.schd.message_queue.put((
-                    args.get('task_job', None),
-                    args.get('event_time', None),
-                    severity, message))
-            return (True, 'Messages queued: %d' % len(messages))
-        if command in ['set_stop_after_clock_time',
-                       'set_stop_after_point',
-                       'set_stop_after_task']:
-            mutate_args = [command, (), {}]
-            for val in args.values():
-                mutate_args[1] = (val,)
-            return self.schd.command_queue.put(tuple(mutate_args))
-        mutate_args = [command, (), {}]
-        for key, val in args.items():
-            if isinstance(val, list):
-                mutate_args[1] = (val,)
-            elif isinstance(val, dict):
-                mutate_args[2] = val
-            else:
-                mutate_args[2][key] = val
-        return self.schd.command_queue.put(tuple(mutate_args))
+    # async def _mutation_mapper(self, command, args):
+    #     """Map between GraphQL resolvers and internal command interface."""
+    #     if command in ['clear_broadcast',
+    #                    'expire_broadcast',
+    #                    'put_broadcast']:
+    #         return getattr(self.schd.task_events_mgr.broadcast_mgr,
+    #                        command, None)(**args)
+    #     if command == 'put_ext_trigger':
+    #         return self.schd.ext_trigger_queue.put((
+    #             args.get('event_message'),
+    #             args.get('event_id')))
+    #     if command == 'put_messages':
+    #         messages = args.get('messages', [])
+    #         for severity, message in messages:
+    #             self.schd.message_queue.put((
+    #                 args.get('task_job', None),
+    #                 args.get('event_time', None),
+    #                 severity, message))
+    #         return (True, 'Messages queued: %d' % len(messages))
+    #     if command in ['set_stop_after_clock_time',
+    #                    'set_stop_after_point',
+    #                    'set_stop_after_task']:
+    #         mutate_args = [command, (), {}]
+    #         for val in args.values():
+    #             mutate_args[1] = (val,)
+    #         return self.schd.command_queue.put(tuple(mutate_args))
+    #     mutate_args = [command, (), {}]
+    #     for key, val in args.items():
+    #         if isinstance(val, list):
+    #             mutate_args[1] = (val,)
+    #         elif isinstance(val, dict):
+    #             mutate_args[2] = val
+    #         else:
+    #             mutate_args[2][key] = val
+    #     return self.schd.command_queue.put(tuple(mutate_args))
+
+    async def _mutation_mapper(self, command, kwargs):
+        # TODO authentication
```

Comment:
Authentication is done by ZmqAuth and Authorisation will be done by GraphQL middleware.

Reply:
This was more a reference to the `user=getpass.getuser()` line below where we assume that any authenticated user *is* the workflow owner. (It's only temporary and no different than the CLI-SuiteRuntimeServer system.)